### ISSUE TITLE
Problem about GNN baseline reproduction（关于GNN baseline的结果复现）

### ISSUE THREAD

**Issue by chengpei-wu** [Starlien95/GraphPrompt#15]
Hello, thank you for your work. 

I have some questions regarding the implementation details of the GNN baselines in the paper.

For the Flickr dataset, I used the 50-shot node classification setting mentioned in the paper (selecting 50 nodes per class for the training and validation sets, with the remaining nodes as the test set). However, under this setting, I was able to easily achieve a **25%** test accuracy using GCN (implemented with the DGL library, with random parameter initialization), which is **significantly higher than the results reported in the paper**. Am I doing something wrong in my experiment?

Could you share your implementation of GCN under few-shot learning?

Thanks!

**Comment by chengpei-wu** [comment]
I also used GCN to perform graph classification in the 5-shot scenario. 

The accuracies of GCN on the PROTEINS, COX2, and ENZYMES datasets are approximately 70%, 79%, and 29%, respectively (while in the paper, they are around 54%, 51%, and 20% in Table 3).

During training, I set the number of epochs fixed to 200, and I used the model from the epoch with the highest accuracy on the validation set to make predictions on the test set as the final result.

**Comment by chengpei-wu** [comment]
Could it be that the author trained the GNN baselines on subgraphs instead of on the original graph like traditional semi-supervised node classification? 

But even if that's the case, the pipeline for graph classification shouldn't change.

**Comment by Starlien95** [comment]
Thank you for your attention to our work. In the few-shot experimental setting, model performance is highly sensitive to the selected training set. In our experiments, we also observed that for different tasks (where we randomly sample k nodes/graphs per class), GraphPrompt and other baselines can achieve very high accuracy on some tasks while performing quite low on others.

Additionally, hyperparameters, random seeds, and the experimental environment can also affect the final results, especially given the limited amount of training data. Therefore, it is not unusual to obtain different results from those reported in the paper.

For a fair comparison, we suggest you testing GraphPrompt under your experimental environment using the same data split as other baselines you tested.

**Comment by chengpei-wu** [comment]
Thank you for your response. The randomness of dataset splitting is indeed very important. 

I will run 100 times of few-shot learning with different splitting and check the final average and standard deviation.
